pkg/roachprod/fluentbit/fluentbit.go

```diff
@@ -0,0 +1,104 @@
+package fluentbit
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"strings"
+	"text/template"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/errors"
+)
+
+//go:embed files/fluent-bit.yaml.tmpl
+var fluentBitTemplate string
+
+//go:embed files/fluent-bit.service
+var fluentBitSystemdUnit string
+
+type Config struct {
+	DatadogSite   string
+	DatadogAPIKey string
+}
+
+func Install(ctx context.Context, l *logger.Logger, c *install.SyncedCluster, config Config) error {
+	if err := install.InstallTool(ctx, l, c, c.Nodes, "fluent-bit", l.Stdout, l.Stderr); err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+	var combinedErr error
+
+	// The Fluent Bit configuration needs the hostname of each node.
+	for _, node := range c.Nodes {
+		buf.Reset()
+
+		if err := c.Run(ctx, l, &buf, l.Stderr, install.WithNodes(install.Nodes{node}), "hostname", "hostname -f"); err != nil {
+			combinedErr = errors.CombineErrors(combinedErr, err)
+			continue
+		}
+
+		data := templateData{
+			DatadogSite:   config.DatadogSite,
+			DatadogAPIKey: config.DatadogAPIKey,
+			Hostname:      strings.TrimSpace(buf.String()),
+			Cluster:       c.Name,
+		}
+
+		fluentBitConfig, err := executeTemplate(data)
+		if err != nil {
+			combinedErr = errors.CombineErrors(combinedErr, err)
+			continue
+		}
+
+		if err := c.PutString(ctx, l, install.Nodes{node}, fluentBitConfig, "/tmp/fluent-bit.yaml", 0644); err != nil {
+			combinedErr = errors.CombineErrors(combinedErr, err)
+			continue
+		}
+	}
+
+	if err := c.PutString(ctx, l, c.Nodes, fluentBitSystemdUnit, "/tmp/fluent-bit.service", 0644); err != nil {
+		combinedErr = errors.CombineErrors(combinedErr, err)
+	}
+
+	if err := c.Run(ctx, l, l.Stdout, l.Stderr, install.WithNodes(c.Nodes), "fluent-bit", `
```

Comment:
The execution speed of this `Install` command could benefit from  `c.ParallelE` since there is no reason to run these sequentially on each node.

Reply:
`Run` documents that commands are run in parallel when running on multiple nodes, which is what's happening here. Happy to change it to `Parallel` or `ParallelE` if that's not the case.

> When running on just one node, the command output is streamed to stdout. When running on multiple nodes, the commands run in parallel, their output is cached and then emitted all together once all commands are completed.

Reply:
Ah sorry, I confused myself with the loop just above that where I think I thought it could all be optimised to run in parallel.

Reply:
I think what I meant is the hostname + config gen could all be grouped with `Parallel` if I'm not mistaken.

Reply:
Moved into `Parallel`. Let me know what you think.

Reply:
Nice Thanks!